File: perl/keys.pl
#!/usr/bin/perl

package current_test;

use strict;
use testenv;

sub println { testenv::println @_}
sub test { testenv::test @_}
sub checkComplainAndAdjustExpected { testenv::checkComplainAndAdjustExpected @_}
sub result { testenv::result @_}

use testdata ;

checkComplainAndAdjustExpected 0;

test "46" == join "", sort keys %testdata::hash;
test "46" == join "", sort keys %{$testdata::hash_ref};

checkComplainAndAdjustExpected 2;

# see also keys.pl, values.pl

println result()

File: clojure1.6/interpreter/keys.clj

(ns keys-test
 (:use testenv)
)

(checkComplainAndAdjustExpected 0)

(teTest (= '(:a :b) (sort (keys {:a 1 :b 2}))))

(teTest (= #{:a :b} (set (keys {:a 1 :b 2}))))

(teTest (= '(:b :a) (keys {:a 1 :b 2}))); ?

(checkComplainAndAdjustExpected 3)

(println (teResult))

; 8b4rxy2qq5 ; see also: topic-list-types.clj topic-map.clj keys.clj

